redis/tests/test-template.sh

```
@@ -183,5 +183,30 @@ assert_contains "tls: cert volume mount" "${tls}" "/etc/redis/tls"
+assert_contains "acl: default user redefined (requirepass dropped)" "${acl}" "user default on"
+assert_contains "acl: replication uses masteruser" "${acl}" "masteruser"
+assert_contains "acl: sentinel auth-user wired" "${acl}" "sentinel auth-user"
+assert_contains "acl: probes carry --user" "${acl}" "redis-cli --user ops"
+
+# Default operator (additive user) keeps the standalone requirepass path; no operator wiring.
+acl_add=$(helm template r "${CHART_DIR}" -f "${SCRIPT_DIR}/values-acl.yaml" 2>&1)
+assert_contains "acl(additive): app user line rendered" "${acl_add}" "user app on"
+assert_not_contains "acl(additive): no masteruser in standalone" "${acl_add}" "masteruser"
+
+# Guard: locking down default without a separate operator is rejected.
+acl_fail=$(helm template r "${CHART_DIR}" --set redis.auth.acl.enabled=true \
```

Shellcheck (0.11.0)

[warning] acl_fail appears unused. Verify use (or export if used externally).

(SC2034)